woocommerce/woocommerce-ios
# Preparations to make AddProductToOrder reusable

Part of #6489 


### Description

As part of the Coupons project, we want to be able to let merchants select products and variations that are applicable or should be excluded for any coupons.

Previously I was trying to update the existing [ProductListSelectorViewController](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/ViewRelated/Products/ProductListSelector/ProductListSelectorViewController.swift), but the work required has been quite tedious. Then I realized that there's an existing view [AddProductToOrder](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/ViewRelated/Orders/Order%20Creation/ProductsSection/AddProductToOrder.swift) that matches quite closely with the design of the Select Products and Exclude Products screen of the Coupon flow. (p1649844987957019-slack-C02TD63FZA8)

The goal now is to make `AddProductToOrder` reusable for more than just the New Order flow but also the Add/Edit Coupon flow. This PR aims to prepare for this update with the following changes:
- Rename `AddProductToOrder` and its view model and unit test to `ProductSelector` to make it more generic.
- Similarly, rename `AddProductVariationToOrder` and its view model and unit test to `ProductVariationSelector` to make it more generic.
- Group these files and `ProductRow` as well as its view model together and move this new group to under `Products` group for easier discovery.
- Move `ProductListSelector` group back to under Products > Edit Product > Linked Product List Selectors. I will revert changes to the Linked Product flow after finishing the updates to `AddProductToOrder`.

### Testing instructions

Please make sure that the add product to order flow still works as before:
- Navigate to Orders tab
- Select "+" button and proceed to "Create order"
- Select "Add Product" button. Add then remove items to see if the functionality still works correctly.

---
